.travis.yml

```diff
@@ -18,26 +18,64 @@ env:
   - DB=mysqli
 
 before_install:
-  - mv ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini{,.disabled} || echo "xdebug not available"
+  - mv ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini{,.disabled} || true
 
 before_script:
   - if [[ "$DB" == "mysql" || "$DB" == "mysqli" || "$DB" == *"mariadb"* ]]; then mysql < tests/travis/create-mysql-schema.sql; fi;
+  - |
+    if [ -v $COVERAGE ]; then
+      mv ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/xdebug.ini{.disabled,} || true
```

Comment:
So we unconditionally disable Xdebug `before_install` and then conditionally enable it here? Wouldn't it be more straightforward to conditionally disable it `before_install` and then conditionally check if the extension is loaded here?

Reply:
But between `before_install` and `before_script`, no PHP scripts are run. So the net effect will be the same.

Reply:
There is _install_ between those two (although it's not ordered accordingy in our config).

* before_install
* install
* before_script
* script

https://docs.travis-ci.com/user/customizing-the-build/#The-Build-Lifecycle

Reply:
But Composer disables Xdebug internally, so it shouldn't affect it.

Reply:
Hmm, is that even possible? Afaik if PHP is started with XDebug registered as zend_extension, this itself has impact on the runtime since XDebug overrides default execution handlers...

Reply:
@Majkl578 see `Composer\XdebugHandler`. It restarts PHP with explicit INI configuration which excludes Xdebug (something like that).

Reply:
I see, it was added in Composer 1.3.0, 15 months ago. Probably safe to assume this behavior now then. :)